scripts/zkai_attention_kv_proof_route_selector_gate.py

```
@@ -3035,6 +3256,86 @@ def validate_d16_quantized_softmax_receipt(summary: Any) -> None:
+    expected_kernel_policy_fields = {
+        "denominator_policy": "sum_positive_statement_bound_table_weights_per_head_step",
+        "division_rule": "output = numerator.div_euclid(denominator); remainder = numerator.rem_euclid(denominator)",
+        "rounding_rule": "floor_toward_negative_infinity_via_euclidean_division_positive_denominator",
+        "head_binding_policy": "each score row binds head_index; outputs are keyed by (head_index, local_step_index)",
+        "step_binding_policy": "each score row binds per-head local step_index derived from statement input_steps order",
+        "causal_mask_policy": "causal_prefix_position_lte_query_token checked on every emitted score row",
+        "weight_table_commitment": "blake2b-256:852c06058232d0c0871d2559e57b55c85ab30932cf07ef1814b01143209706f0",
+    }
+    for key, expected_value in expected_kernel_policy_fields.items():
+        if summary.get(key) != expected_value:
+            raise AttentionKvRouteSelectorError(f"d16 two-head quantized Softmax {key} drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)